src/Riok.Mapperly/Symbols/GetterMemberPath.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Riok.Mapperly.Descriptors;
+using Riok.Mapperly.Emit.Syntax;
+using Riok.Mapperly.Helpers;
+
+namespace Riok.Mapperly.Symbols;
+
+public class GetterMemberPath : MemberPath
+{
+    private GetterMemberPath(IReadOnlyList<IMappableMember> path)
+        : base(path) { }
+
+    public static GetterMemberPath Build(MappingBuilderContext ctx, MemberPath memberPath)
+    {
+        var path = memberPath.Path;
+        var array = path.Select(item =>
+            {
```

Comment:
IMO the readability could be improved if this is extracted into an own static method instead of a lambda.